### PR TITLE
Update DG, FindXCommand, Assign and Unaasign

### PR DESCRIPTION
**Pull request by wanyu-l**
* added manual test cases for interviews to DG
* updated FindXCommands' display message for optional fields
* updated Assign and Unassign to display error message when index for interview is missing

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -554,3 +554,111 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+
+## **Appendix: Instructions for manual testing**
+
+###Interview Management
+
+####Add an Interview
+
+1. Adding an interview without any candidates to HR Manager<br>
+   i. Prerequisites: Position exists in HR Manager.
```
Comment (seaweediman) [inline docs/DeveloperGuide.md:566]: Should mention that Position Bookkeeper exists....

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -554,3 +554,111 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
 * **Private contact detail**: A contact detail that is not meant to be shared with others
+
+
+## **Appendix: Instructions for manual testing**
+
+###Interview Management
```
Comment (seaweediman) [inline docs/DeveloperGuide.md:561]: Need to add spacing between ### and your headings

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -554,3 +554,111 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+   Expected: new interview for `Bookkeeper` on `15 Oct 2021`, `1400 ~ 1600` with status `PENDING` is added to the bottom of the list of interviews.<br>
+
+2. Adding an interview with first candidate to HR Manager<br>
+   i. Prerequisites: Position exits in HR Manager, current displayed list of candidates is not empty and candidate with corresponding index has applied for position.<br>
+   ii. Test case: `add_i position=Bookkeeper c=1 date=15/10/2021 time=1400 duration=120 interviewed=pending`<br>
+   Expected: new interview for `Bookkeeper` on `15 Oct 2021`, `1400 ~ 1600` with status `PENDING`, with first candidate from the currently displayed list of candidates is added to the bottom of the list of interviews.<br>
+
+3. Adding an interview with multiple candidates to HR Manager<br>
+   i. Prerequisites: Position exits in HR Manager, current displayed list of candidates has more than 1 entry and candidates with corresponding indexes has applied for position.<br>
+   ii. Test case: `add_i position=Bookkeeper c=1 2 date=15/10/2021 time=1400 duration=120 interviewed=pending`<br>
```
Comment (seaweediman) [inline docs/DeveloperGuide.md:577]: Won't this test case conflict test case 2? Cause they are the same interview right?

Comment (wanyu-l) [reply docs/DeveloperGuide.md:590]: I was thinking that they are alternative test cases but sure I'll change it

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -554,3 +554,139 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+   * No other interview for `Bookkeeper` with the same date, time and duration that already exist in HR Manager.
+   * Current displayed list of candidates has at least 2 entries and the first and second candidates have applied for `Bookkeeper`.<br>
+    
+   ii. Test case: `add_i position=Bookkeeper c=1 2 date=17/10/2021 time=1400 duration=120 interviewed=pending`<br>
+   Expected: new interview for `Bookkeeper` on `17 Oct 2021`, `1400 ~ 1600` with status `PENDING`, with first and second candidates from the current displayed list of candidates is added to the bottom of the list of interviews.<br>
+   
+
+4. Incorrect Add Interview Commands<br>
+   i. Prerequisites: `Accountant` does not exist in HR Manager but `Bookkeeper` does.
+   i. Test case: `add_i`<br>
```
Comment (seaweediman) [inline docs/DeveloperGuide.md:596]: 2 i.

File: src/main/java/seedu/address/model/interview/Interview.java [hunk 5]
```diff
@@ -205,6 +206,19 @@ public void setStatus(InterviewStatus status) {
...
+     * @return true if they start at the same time or if one begins before the other ends.
+     */
+    public boolean hasOverLapWith(Interview other) {
+        LocalDateTime start = LocalDateTime.of(getDate(), getStartTime());
+        LocalDateTime end = LocalDateTime.of(getDate(), getEndTime());
+        LocalDateTime toScheduleStart = LocalDateTime.of(other.getDate(), other.getStartTime());
+        return toScheduleStart.isAfter(start) && toScheduleStart.isBefore(end)
+                || start.isEqual(toScheduleStart);
+    }
+
```
Comment (seaweediman) [inline src/main/java/seedu/address/model/interview/Interview.java:221]: Let's say c1 already has i1 : 1300-1500

If assign c1 to i2: 1200-1400, wouldn't this be allowed? Cause 1200 !> 1300 and 1200<1400

Comment (wanyu-l) [reply src/main/java/seedu/address/model/interview/Interview.java:236]: yep, just updated it for a bidirectional check

Comment (wanyu-l) [reply src/main/java/seedu/address/model/interview/Interview.java:236]: oh sorry i misunderstood your comment, the getHour() checks are only for interviews that go beyond 1 "day", to update the variable with correct LocalDateTime coz we are not creating the "other" LocalDate object anywhere

**Review by JeffZincatz — APPROVED** [review]
LGTM.

**Review by sueann-chua — APPROVED** [review]
looks good

**Pull request merged**